src/codegen/util/hash_table.cpp

```diff
@@ -203,7 +203,7 @@ void HashTable::BuildLazy() {
   // TODO: Use sketches to estimate the real # of unique elements
   // TODO: Perhaps change probing strategy based on estimate?
 
-  directory_size_ = NextPowerOf2(num_elems_) * 2;
+  directory_size_ = (num_elems_ == 0) ? 0 : NextPowerOf2(num_elems_) * 2;
```

Comment:
The problem with this is that you've overflowed `directory_mask_` when no elements exist.

Leave this as is and put an early exit at the top of the function. At this point, the hash table is still valid (i.e., has a directory, mask, size, capacity etc.). So, probes shouldn't find anything.

Reply:
I thought about that and decided the mask is valid after that - but's you are right, it is not.